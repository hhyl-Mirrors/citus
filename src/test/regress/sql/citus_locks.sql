-- store the current gpid in a variable
SELECT global_pid AS coordinator_gpid
FROM get_all_active_transactions()
WHERE process_id = pg_backend_pid() \gset

-- list the locks on relations for current distributed transaction
SELECT relation::regclass, nodeid, mode, granted
FROM citus_locks
WHERE global_pid = :coordinator_gpid AND locktype = 'relation'
ORDER BY 1, 2, 3, 4;
