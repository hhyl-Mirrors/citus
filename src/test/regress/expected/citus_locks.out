-- store the current gpid in a variable
SELECT global_pid AS coordinator_gpid
FROM get_all_active_transactions()
WHERE process_id = pg_backend_pid() \gset
-- list the locks on relations for current distributed transaction
SELECT relation::regclass, nodeid, mode, granted
FROM citus_locks
WHERE global_pid = :coordinator_gpid AND locktype = 'relation'
ORDER BY 1, 2, 3, 4;
              relation              | nodeid |      mode       | granted
---------------------------------------------------------------------
 pg_locks                           |      2 | AccessShareLock | t
 pg_dist_partition                  |      2 | AccessShareLock | t
 pg_dist_partition                  |      2 | AccessShareLock | t
 pg_dist_node                       |      2 | AccessShareLock | t
 pg_dist_node_pkey                  |      2 | AccessShareLock | t
 pg_dist_node_nodename_nodeport_key |      2 | AccessShareLock | t
 pg_dist_local_group                |      2 | AccessShareLock | t
 pg_dist_local_group_pkey           |      2 | AccessShareLock | t
 citus_locks                        |      2 | AccessShareLock | t
(9 rows)

